networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -91,12 +133,21 @@ def precheck(G1, G2, G1_labels, G2_labels):
 
     G1_labels,G2_labels: dict
         The label of every node in G1 and G2 respectively.
+
+        node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
     """
     if G1.order() != G2.order():
         return False
     if sorted(d for n, d in G1.degree()) != sorted(d for n, d in G2.degree()):
         return False
 
+    G1_labels.update(G1.nodes(data=node_labels, default=default_label))
+    G2_labels.update(G2.nodes(data=node_labels, default=default_label))
+
```

Comment:
Should `_precheck` create the two dicts and return them?  Is there every a case when non-empty dicts should be passed into this function? It's a style issue more than anything, so you get to choose. What do you think?

Reply:
Every time, precheck computes the labels so there is no possibility to pass non-empty labels to precheck. Yea we could make it return them, however we have to figure out how to restructure the usage of precheck in vf2++ so it is not ugly. Specifically, right now we are doing

```python
if not _precheck(G1, G2, G1_labels, G2_labels, node_labels, default_label):
    return False
```
If precheck returns the labels as well, we will have to do smth like this probably
```python
checkspass, labels1, labels2 = precheck(...)
if not checkpass:
  return False
```
Do we want that?

Reply:
No -- I agree. We don't want that.  I forgot that it returned True/False...  
Leave it the way it is -- unless a better way is suggested by someone else. 
This is good.